lib/react/rails/engine.rb

```diff
@@ -2,7 +2,11 @@ module React
   module Rails
     class Engine < ::Rails::Engine
       initializer "react_rails.setup_engine", :group => :all do |app|
-        app.assets.register_engine '.jsx', React::JSX::Template
+        if app.assets.nil?
+          Sprockets.register_engine  '.jsx', React::JSX::Template
+        else
+          app.assets.register_engine '.jsx', React::JSX::Template
+        end
```

Comment:
I also like the idea of removing that duplication, maybe:

```ruby 
sprockets_env = app.assets || Sprockets # Sprockets 3.x expects this in a different place 
sprockets_env.register_engine(".jsx", React::JSX::Template)
```

IMO less duplication = easier to maintain in the future!